apps/gp2-server/src/handlers/event/algolia-index-calendar-handler.ts

```diff
@@ -0,0 +1,84 @@
+import { AlgoliaClient, algoliaSearchClientFactory } from '@asap-hub/algolia';
+import { gp2 as gp2Model, ListResponse } from '@asap-hub/model';
+import {
+  loopOverCustomCollection,
+  LoopOverCustomCollectionFetchOptions,
+} from '@asap-hub/server-common';
+import { EventBridgeEvent } from 'aws-lambda';
+import { algoliaApiKey, algoliaAppId, algoliaIndex } from '../../config';
+import EventController from '../../controllers/event.controller';
+import { EventContentfulDataProvider } from '../../data-providers/event.data-provider';
+import {
+  getContentfulGraphQLClientFactory,
+  getContentfulRestClientFactory,
+} from '../../dependencies/clients.dependency';
+import logger from '../../utils/logger';
+import { sentryWrapper } from '../../utils/sentry-wrapper';
+import { CalendarPayload } from '../event-bus';
+
+export const indexCalendarEventsHandler =
+  (
+    eventController: EventController,
+    algoliaClient: AlgoliaClient<'gp2'>,
+  ): ((
+    event: EventBridgeEvent<gp2Model.CalendarEvent, CalendarPayload>,
+  ) => Promise<void>) =>
+  async (event) => {
+    logger.debug(`Event ${event['detail-type']}`);
+
+    const fetchFunction = ({
+      skip,
+      take,
+    }: LoopOverCustomCollectionFetchOptions): Promise<
+      ListResponse<gp2Model.EventResponse>
+    > =>
+      eventController.fetch({
+        skip,
+        take,
+        filter: { calendarName: event.detail.resourceId },
```

Comment:
I thought the `event.detail.resourceId` would be the id of the calendar not the calendar name?